client/state/sites/products/actions.js

```diff
@@ -40,29 +40,26 @@ export function fetchSiteProducts( siteId ) {
 			siteId,
 		} );
 
-		return new Promise( ( resolve ) => {
-			wpcom.undocumented().getSiteProducts( siteId, ( error, data ) => {
-				if ( error ) {
-					debug( 'Fetching site products failed: ', error );
+		return wpcom.req
+			.get( `/sites/${ siteId }/products` )
```

Comment:
It's nice that because `siteId` is always a number, we don't need to `encodeURIComponent` like the original method did.